src/parser/shaman/elemental/modules/talents/CallTheThunder.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+
+import SPELLS from 'common/SPELLS';
+import { formatNumber } from 'common/format';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+
+import { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import SpellLink from 'common/SpellLink';
+import Events from 'parser/core/Events';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import { TooltipElement } from 'common/Tooltip';
+
+class CallTheThunder extends Analyzer {
+  overcapPrevented = 0;
+  maelstromAtPreviousEnergize = 0;
+
+  earthShockCasts = 0;
+  earthQuakeCasts = 0;
+
+  earthShockDamage = 0;
+  earthQuakeDamage = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.CALL_THE_THUNDER_TALENT.id);
+
+    this.addEventListener(Events.energize.by(SELECTED_PLAYER), this.onEnergize);
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EARTH_SHOCK), this.onEarthShockOrEarthquakeCast);
```

Comment:
Kind of defeating the purpose if you have a eventlistener point to somethign then still have to do the spellId check. Make another method for one of the spells